src/util.js

```diff
@@ -215,21 +215,63 @@ function round(value, decimalPlaces) {
   return Math.round(value * factor) / factor;
 }
 
-function objAsString(value) {
+function objAsString(obj) {
+  // Get a string representation of a simple object
   let result = '';
-  for (let key in value) {
+  for (let key in obj) {
     if (result) {
-      result += `, ${key}: ${value[key]}`;
+      result += `, ${key}: ${obj[key]}`;
     } else {
       // This is the first key, value pair being printed
-      result += `${key}: ${value[key]}`;
+      result += `${key}: ${obj[key]}`;
     }
   }
   return result;
 }
 
+function cosineDeclinationTerm(declination) {
+  let cosDec = Math.cos((declination * Math.PI) / 180);
+  // If the cosine dec ends up being 0, offset it slightly so that there are no divisions by zero. It doesn't need to be that
+  // precise since this is only for visualization purposes and a small shift won't matter.
+  cosDec = Math.max(cosDec, 10e-4);
+  return cosDec;
+}
+
+function offsetCoordinate(initial, offset) {
+  // Calculate coordinates from provided offsets. Equations pulled
+  // from https://www.atnf.csiro.au/computing/software/miriad/doc/offset.html
+  // Takes a object with the initial coordinate and an object with the offsets,
+  // where each object has "ra" and "dec" keys, and outputs an object with the
+  // initial coordinate with the offset applied.
+  const ARC_SEC_PER_DEG = 3600;
+  let finalDec = initial['dec'] + offset['dec'] / ARC_SEC_PER_DEG;
+  return {
+    ra: initial['ra'] + offset['ra'] / ARC_SEC_PER_DEG / cosineDeclinationTerm(finalDec),
+    dec: finalDec
+  };
+}
+
+function rotateCoordinate(coordinate, center, angle) {
+  // Move the target coordinate to the origin. RA goes along the x direction and Declination along the y.
```

Comment:
I think we may always want to rotate about the origin.. is so maybe we should just remove the center argument here and simplify the function. Then rotation should always be used before offsetting.

Reply:
Yeah it's true we are currently only using this to rotate about the origin... I'll set `center` to default to the origin which will simplify all the places it's used.